src/main/webapp/app/exercises/shared/rating/rating.component.ts

```diff
@@ -46,6 +44,8 @@ export class RatingComponent implements OnInit {
 
         // update feedback locally
         this.rating.rating = $event.newValue;
+        // Delete participation to avoid circular dependency
+        this.rating.result!.participation = undefined;
```

Comment:
I am personally not a fan of the exclamation mark as it says "assume the property/variable **IS** defined", but what if not? There must be a reason why the result can be undefined, so I´d rather go with something like this here:
```suggestion
        if (this.rating.result) {
                this.rating.result.participation = undefined;
        }
```

Reply:
Done in `a92ca57`